src/components/core/GeoChart.tsx

```diff
@@ -0,0 +1,112 @@
+// (C) 2019-2020 GoodData Corporation
+import * as React from "react";
+import { WrappedComponentProps } from "react-intl";
+import * as invariant from "invariant";
+
+import { ICommonChartProps } from "./base/BaseChart";
+import { BaseVisualization } from "./base/BaseVisualization";
+import {
+    commonDefaultProps,
+    ILoadingInjectedProps,
+    visualizationLoadingHOC,
+} from "./base/VisualizationLoadingHOC";
+import GeoChartLegendRenderer, { IChartLegendProps } from "./geoChart/GeoChartLegendRenderer";
+import GeoChartRenderer, { IGeoChartRendererProps } from "./geoChart/GeoChartRenderer";
+
+import { IDataSourceProviderInjectedProps } from "../afm/DataSourceProvider";
+import { DEFAULT_DATA_POINTS_LIMIT } from "../../constants/geoChart";
+import { IGeoConfig, IGeoDataIndex } from "../../interfaces/GeoChart";
+import { getGeoDataIndex, isDataOfReasonableSize } from "../../helpers/geoChart";
+
+export function renderChart(props: IGeoChartRendererProps): React.ReactElement {
+    return <GeoChartRenderer {...props} />;
+}
+
+export function renderLegend(props: IChartLegendProps): React.ReactElement {
+    return <GeoChartLegendRenderer {...props} />;
+}
+
+export interface ICoreGeoChartProps extends ICommonChartProps, IDataSourceProviderInjectedProps {
+    config?: IGeoConfig;
+    chartRenderer?: (props: IGeoChartRendererProps) => React.ReactElement;
+    legendRenderer?: (props: IChartLegendProps) => React.ReactElement;
+}
+
+export type IGeoChartInnerProps = ICoreGeoChartProps &
+    ILoadingInjectedProps &
+    IDataSourceProviderInjectedProps &
+    WrappedComponentProps;
+
+/**
+ * Geo Chart react component
+ */
+export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, {}> {
+    public static defaultProps: Partial<IGeoChartInnerProps> = {
+        ...commonDefaultProps,
+        chartRenderer: renderChart,
+        legendRenderer: renderLegend,
+    };
+
+    public componentDidUpdate(prevProps: IGeoChartInnerProps) {
+        if (this.props.execution !== prevProps.execution) {
```

Comment:
"this.props.execution !== prevProps.execution" => can be wrong.

Reply:
fixed